backend/internal/storage/postgres/db.go

```diff
@@ -4,25 +4,40 @@ import (
 	"context"
 	"fmt"
 	"github.com/CamPlume1/khoury-classroom/internal/config"
-	"github.com/jackc/pgx/v5"
+	"github.com/CamPlume1/khoury-classroom/internal/errs"
+	"github.com/jackc/pgx/v5/pgxpool"
 )
 
 type DB struct {
-	conn *pgx.Conn
+	connPool *pgxpool.Pool
 }
 
 
-// ConnectSupabaseDB establishes a connection and returns it for querying.
+
+// Establishes a postgres connection pool and returns it for querying.
 func New(ctx context.Context, config config.Database) (*DB, error) {
-	conn, err := pgx.Connect(ctx, config.URL)
+	connPool, err := pgxpool.New(ctx, config.URL)
 	if err != nil {
 		fmt.Println(err)
 		return nil, err
 	}
-	fmt.Println("Successfully connected to the database!")
-	return &DB{conn: conn}, nil
+	
+  fmt.Println("Successfully connected to the database!")
+	return &DB{connPool: connPool}, nil
 }
 
+// Closes a connection pool, pings to confirm closure
 func (db *DB) Close(ctx context.Context) error {
-	return db.conn.Close(ctx)
+	db.connPool.Close()
+
+  err := db.connPool.Ping(ctx)
```

Comment:
Given Ping and Close don't actually give the user any info besides success/failure, IMO we should remove as an argument and just use the background context in these methods

Reply:
Gotcha, to be clear this is implies we just close and return nothing?